lib/internal/util/parse_args/parse_args.js

```diff
@@ -304,8 +326,22 @@ const parseArgs = (config = kEmptyObject) => {
         }
       }
 
+      const multipleOption = objectGetOwn(optionConfig, 'multiple');
       if (ObjectHasOwn(optionConfig, 'multiple')) {
-        validateBoolean(optionConfig.multiple, `options.${longOption}.multiple`);
+        validateBoolean(multipleOption, `options.${longOption}.multiple`);
+      }
+
+      const defaultValue = objectGetOwn(optionConfig, 'default');
+      if (defaultValue !== undefined) {
+        if (optionType === 'string' && !multipleOption) {
+          validateString(defaultValue, `options.${longOption}.default`);
+        } else if (optionType === 'string' && multipleOption) {
+          validateStringArray(defaultValue, `options.${longOption}.default`);
+        } else if (optionType === 'boolean' && !multipleOption) {
+          validateBoolean(defaultValue, `options.${longOption}.default`);
+        } else if (optionType === 'boolean' && multipleOption) {
+          validateBooleanArray(defaultValue, `options.${longOption}.default`);
+        }
```

Comment:
nit: we could avoid the code repetition here.
```suggestion
        let validator;
        switch(optionType) {
          case 'string':
          validator = multipleOption ? validateStringArray : validateString;
          break;
          
         case 'boolean':
         validator = multipleOption ? validateBooleanArray : validateBoolean;
         break;
       }
       validator(defaultValue, `options.${longOption}.default`);
```